sqlglot/dialects/oracle.py

```diff
@@ -370,3 +430,25 @@ def into_sql(self, expression: exp.Into) -> str:
                 return f"{self.seg(into)} {self.sql(expression, 'this')}"
 
             return f"{self.seg(into)} {self.expressions(expression)}"
+
+        def hint_sql(self, expression: exp.Hint) -> str:
+            expressions = []
+
+            for expression in expression.expressions:
+                if isinstance(expression, exp.Anonymous):
+                    formatted_args = self._format_hint_function_args(*expression.expressions)
+                    expressions.append(f"{self.sql(expression, 'this')}({formatted_args})")
+                else:
+                    expressions.append(self.sql(expression))
+
+            return f" /*+ {self.expressions(sqls=expressions, sep=self.QUERY_HINT_SEP).strip()} */"
+
+        def _format_hint_function_args(self, *args: t.Optional[str | exp.Expression]) -> str:
+            arg_sqls = tuple(
+                self.sql(arg) for arg in args if arg is not None and not isinstance(arg, bool)
```

Comment:
Why would `arg` be either `None` or `bool` here? Is this copy-pasted from `format_args`? What not reuse that?

Reply:
@georgesittas Yes this was copy-pasted from format_args.

We cannot reuse it as is because `format_args` joins with `', '`/`',\n'` instead of `' '`/`'\n'`.

Actually my code has a bug, it will pretty print with comma instead of space when the hint function arguments are too long.

I've fixed the bug and added a test case, as well as changed the args_sql to `arg_sqls = tuple(self.sql(arg) for arg in args)`

Reply:
I see, I may try to refactor this later so that we can consolidate the two implementations somehow.